src/main/java/net/sf/jabref/gui/EntryCustomizationDialog.java

```diff
@@ -376,10 +378,11 @@ public void actionPerformed(ActionEvent e) {
      * a valid type, that no obsolete entry editors are around, and that
      * the right-click menus' change type menu is up-to-date.
      */
-    private void updateTypesForEntries(String typeName) {
+    private void updateTypesForEntries() {
         for (BasePanel bp : frame.getBasePanelList()) {
 
             for (BibEntry entry : bp.getDatabase().getEntries()) {
+                // Please explain what this line, and therefore the whole method, is doing (if it is doing something)
                 EntryTypes.getType(entry.getType(), bibDatabaseMode).ifPresent(entry::setType);
```

Comment:
I think, it is because the types are objects. If one switches from bibtex to biblatex, the types get different objects even though the string representation is the same. This loop ensures that the type objects match the type of the database.